examples/grpc-client/client.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/coreos/dex/api"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+func newDexClient(hostAndPort, caPath, clientCrt, clientKey string) (api.DexClient, error) {
+	cPool := x509.NewCertPool()
+	caCert, err := ioutil.ReadFile(caPath)
+	if err != nil {
+		return nil, fmt.Errorf("invalid CA crt file: %s", caPath)
+	}
+	if cPool.AppendCertsFromPEM(caCert) != true {
+		return nil, fmt.Errorf("failed to parse CA crt")
+	}
+
+	clientCert, err := tls.LoadX509KeyPair(clientCrt, clientKey)
+	if err != nil {
+		return nil, fmt.Errorf("invalid client crt file: %s", caPath)
+	}
+
+	clientTLSConfig := &tls.Config{
+		RootCAs:      cPool,
+		Certificates: []tls.Certificate{clientCert},
+	}
+	creds := credentials.NewTLS(clientTLSConfig)
+
+	conn, err := grpc.Dial(hostAndPort, grpc.WithTransportCredentials(creds))
+	if err != nil {
+		return nil, fmt.Errorf("dail: %v", err)
+	}
+	return api.NewDexClient(conn), nil
+}
+
+func main() {
+	var caCrt = flag.String("ca_crt", "", "CA certificate")
```

Comment:
nit: we generally use "-" in flags instead of "_"

Also the `var` is redundant. You can just do "caCert := flag.String"